includes/class-amp-theme-support.php

```diff
@@ -1760,19 +1788,36 @@ public static function prepare_response( $response, $args = array() ) {
 		$current_url = amp_get_current_url();
 		$ampless_url = amp_remove_endpoint( $current_url );
 
+		// When response caching is enabled, determine if it should be turned off for cache misses.
+		$caches_for_url                 = null;
+		self::$post_processor_cache_key = null;
+		if ( true === $args['enable_response_caching'] ) {
+			self::$post_processor_cache_key = md5( $current_url );
+			$caches_for_url                 = wp_cache_get( self::$post_processor_cache_key, self::POST_PROCESSOR_CACHE_EFFECTIVENESS );
```

Comment:
Sone of the subsequent checks could perhaps be simplified by checking if this is an array here, and if not, just set it to be an empty array.